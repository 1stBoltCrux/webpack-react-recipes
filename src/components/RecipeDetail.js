import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom'

const RecipeDetail = (props) => {
  const { recipe, style, className } = props;
  if (recipe) {
    return (
      <div className="pl4 p2 bg-white">
        <h2>{recipe.name}</h2>
        <img
          className="fit"
          src={recipe.image}
          alt={recipe.name}
        />
        <div>
          <span>{recipe.category}</span>
          <span>{recipe.calories}</span>
        </div>
        <h3>Ingredients</h3>
        {recipe.ingredients &&
          <ul>
            {recipe.ingredients.map(ingredient => (
              <li key={ingredient}>
                {ingredient}
              </li>
            ))}
          </ul>

        }

        <h3>Steps</h3>
        {props.recipe.steps &&
          <ol>{props.recipe.steps.map(step => <li key={step}>{step}</li>)}</ol>
        }
          <Link to={`/recipe/${props.recipe.id}`}>See More</Link>

      </div>
    );
  }
  return (
    <p
      style={style}
      className={classNames(`h3 p2 bg-white italic center ${className}`)}
    >
Please select a recipe to view detail
    </p>
  );
};

export default RecipeDetail;

RecipeDetail.propTypes = {
  style: PropTypes.object,
  recipe: PropTypes.object,
  className: PropTypes.string,
};
