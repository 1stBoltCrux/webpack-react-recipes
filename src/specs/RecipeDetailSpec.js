import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import RecipeDetail from '../components/RecipeDetail';

const testRecipe = {
  id: 1,
  name: 'Test Recipe',
  category: 'Test Category',
  ingredients: ['ing1', 'ing2'],
  steps: ['step1', 'step2']
}

describe('<RecipeDetail />', () => {

  let testRecipe;

  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'Test Recipe',
      category: 'Test Category',
      ingedients: ['ing1', 'ing2'],
      steps: ['step1', 'step2']
    }
  })
  test('Should render zero state', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail/>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('Should correctly render a recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe}/>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('Should correctly assign a classnName', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} className='classname-test'/>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
  test('Should render a recipe without ingredients', () => {
    delete testRecipe.ingedients
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe}/>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
  test('Should render a recipe without steps', () => {
    delete testRecipe.steps
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail/>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
})
