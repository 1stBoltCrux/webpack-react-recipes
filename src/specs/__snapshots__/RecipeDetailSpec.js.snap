// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> Should correctly assign a classnName 1`] = `
<div
  className="pl4 p2 bg-white"
>
  <h2>
    Test Recipe
  </h2>
  <img
    alt="Test Recipe"
    className="fit"
  />
  <div>
    <span>
      Test Category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      step1
    </li>
    <li>
      step2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> Should correctly render a recipe 1`] = `
<div
  className="pl4 p2 bg-white"
>
  <h2>
    Test Recipe
  </h2>
  <img
    alt="Test Recipe"
    className="fit"
  />
  <div>
    <span>
      Test Category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      step1
    </li>
    <li>
      step2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> Should render a recipe without ingredients 1`] = `
<div
  className="pl4 p2 bg-white"
>
  <h2>
    Test Recipe
  </h2>
  <img
    alt="Test Recipe"
    className="fit"
  />
  <div>
    <span>
      Test Category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      step1
    </li>
    <li>
      step2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> Should render a recipe without steps 1`] = `
<p
  className="h3 p2 bg-white italic center undefined"
>
  Please select a recipe to view detail
</p>
`;

exports[`<RecipeDetail /> Should render zero state 1`] = `
<p
  className="h3 p2 bg-white italic center undefined"
>
  Please select a recipe to view detail
</p>
`;
